test/modules/CloudToDeviceMessageTester/CloudToDeviceMessageSender.cs

```diff
@@ -58,36 +58,37 @@ public async Task StartAsync(CancellationToken ct)
 
             while (!ct.IsCancellationRequested && this.IsTestTimeUp(testStartAt))
             {
-                MessageTestResult testResult = await this.SendCloudToDeviceMessageAsync(batchId, this.trackingId);
-                await ModuleUtil.ReportTestResultAsync(this.testResultReportingClient, this.logger, testResult);
-                this.messageCount++;
-                await Task.Delay(this.messageDelay, ct);
+                MessageTestResult testResult = null;
+                try
+                {
+                    testResult = await this.SendCloudToDeviceMessageAsync(batchId, this.trackingId);
+                    this.messageCount++;
+                    await ModuleUtil.ReportTestResultAsync(this.testResultReportingClient, this.logger, testResult);
+                    await Task.Delay(this.messageDelay, ct);
+                }
+                catch (Exception ex)
```

Comment:
@philipktlin  Should this be catching a specific exception for full message queue and rethrow the rest of the exception? Or should it just swallow all the exception here?

Reply:
Changed to catch specific exception, log and swallow

Also will catch general exception, but will log and re-throw.